pyth-sdk-solana/src/state.rs

```diff
@@ -323,38 +345,38 @@ unsafe impl Pod for PriceAccount {
 
 impl PriceAccount {
     pub fn to_price_feed(&self, price_key: &Pubkey) -> PriceFeed {
-        let mut status = self.agg.status;
-        let mut prev_price = self.prev_price;
-        let mut prev_conf = self.prev_conf;
-        let mut prev_publish_time = self.prev_timestamp;
-
-        if let Ok(clock) = Clock::get() {
-            if matches!(status, PriceStatus::Trading)
-                && clock.slot.saturating_sub(self.agg.pub_slot) > VALID_SLOT_PERIOD
-            {
-                status = PriceStatus::Unknown;
-                prev_price = self.agg.price;
-                prev_conf = self.agg.conf;
-                prev_publish_time = self.timestamp;
+        let status = self.agg.status;
+
+        let price = if matches!(status, PriceStatus::Trading) {
+            Price {
+                conf:         self.agg.conf,
+                expo:         self.expo,
+                price:        self.agg.price,
+                publish_time: self.timestamp,
+            }
+        } else {
+            Price {
+                conf:         self.prev_conf,
+                expo:         self.expo,
+                price:        self.prev_price,
+                publish_time: self.prev_timestamp,
             }
+        };
+
+        let mut ema_price = Price {
+            conf:         self.ema_conf.val as u64,
+            expo:         self.expo,
+            price:        self.ema_price.val,
+            publish_time: self.timestamp,
+        };
+
+        // Ema price only gets updated if status is Trading. So it's update timestamp will be
+        // prev_timestamp when the status is not Trading.
+        if !matches!(status, PriceStatus::Trading) {
+            ema_price.publish_time = self.prev_timestamp;
         }
 
```

Comment:
We effectively use the same timestamp for both EMA and non-EMA prices, so can we abstract this away behind a `get_timestamp` function, to make this clear? Instead of doing the same timestamp logic twice.

Reply:
I don't have strong opinion. I think it's fine to keep it as it is. This logic is only used here and I don't see much added value of abstracting it.

Reply:
The motivation for this came from increasing readability, not reducing duplication. It's not obvious from skimming the code that the timestamps are the same, using the same variable/function for them would make this very clear.

Reply:
Hmm... got it.